webapp/crashstats/signature/views.py

```diff
@@ -84,17 +105,8 @@ def signature_report(request, params, default_context=None):
 
     context["signature"] = signature
 
-    fields = sorted(
-        x["name"]
-        for x in SuperSearchFields().get().values()
-        if x["is_exposed"]
-        and x["is_returned"]
-        and request.user.has_perms(x["permissions_needed"])
-        and x["name"] != "signature"  # exclude the signature field
-    )
-    context["fields"] = [
-        {"id": field, "text": field.replace("_", " ")} for field in fields
-    ]
+    fields = get_fields(request.user)
+    context["fields"] = fields
 
     columns = request.GET.getlist("_columns")
     columns = [x for x in columns if x in fields]
```

Comment:
fields now contains dictionaries instead of strings, but it looks here like it's being checked for containing strings? and then the test still passes because the default columns are being used when nothing matches?

Reply:
Oh, whoops--this is a mistake. It's not related to the test because the test only tests `get_fields()` by itself.